operate/webapp/src/main/java/io/camunda/operate/webapp/backup/BackupRepository.java

```diff
@@ -23,4 +30,25 @@ public interface BackupRepository {
 
   void executeSnapshotting(
       BackupService.SnapshotRequest snapshotRequest, Runnable onSuccess, Runnable onFailure);
+
+  default boolean isIncompleteCheckTimedOut(
+      final OperateProperties operateProperties,
+      final long startTimeInMilliseconds,
+      final long endTimeInMilliseconds) {
+    try {
+      final var now = Instant.now().toEpochMilli();
+      final var timeout = operateProperties.getBackup().getIncompleteCheckTimeoutInSeconds();
+      // if is no end time available we just check start time with now
+      if (now - startTimeInMilliseconds > timeout && endTimeInMilliseconds <= 0) {
```

Comment:
I don't understand the need for this check: endTimeInMilliseconds <= 0

Reply:
By default endTimeInMilliseconds is set to 0. That means it is not set or available. I'll add a comment.